contribution/extension/type.php

```diff
@@ -337,6 +349,49 @@ protected function update_phpbb_requirement(array $data)
 	}
 
 	/**
+	 * Checks if the version is stable (1.2.3) or not (1.2.3-RC1)
+	 *
+	 * @param string $version
+	 * @return bool
+	 */
+	protected function is_stable_version($version)
+	{
+		$version = strtolower($version);
+
+		$unstable = array(
+			'rc',
+			'alpha',
+			'beta',
+			'dev',
+			'a',
+			'b',
+		);
+
+		$option = implode('|', $unstable);
+
+		// Check the version format and numbering
+		if (preg_match('#((\d+)\.)+(\d+)[a-z]?#', $version, $matches))
```

Comment:
I think we only allow versions like `x.x.x` and `x.x.x-PLx` so it may be better to just test for that, rather than try to guess what unstable versions might be with a fixed group of them (people may invent others). 

Might be able to simplify this whole function down to just:

```php
return (preg_match('#\d\.\d+\.\d+(-pl\d+)?#i', $version) && version_compare('1.0.0', $version) > 0);
```

Even if you don't do this, your existing code does not need `$matches` since you don't use it.

Reply:
You're right, checking for allowed versions is easier. I had to leave the `$matches` in there because `version_compare()` would report a `-PLx` version as being `< 1.0.0`. Only `x.y.z` should be passed to it.